common/wuanalysis/anawu.cpp

```diff
@@ -95,15 +95,100 @@ class WuScope : public CInterface, implements IWuEdge, implements IWuActivity
 };
 
 //-----------------------------------------------------------------------------------------------------------
+typedef enum {wutOptValueTypeFirst=0, wutOptValueTypeMSec=0, wutOptValueTypeSeconds, wutOptValueTypePercent, wutOptValueTypeCount, wutOptValueTypeMax} WutOptValueType;
+
+struct WuOption
+{
+    const char * name;
+    stat_type defaultValue;
+    WutOptValueType type;
+};
+
+constexpr struct WuOption wuOptionsDefaults[watOptMax]
+= { [watOptMinInterestingTime] = { "minInterestingTime", 1000, wutOptValueTypeMSec},
```

Comment:
Unfortunately I think this is a C99 feature, rather than c++11, so we can't use it.

Reply:
You could add an extra entry to the option, and check that i-1 == option[i-1].option